config.py

```diff
@@ -9,119 +15,82 @@
 __author__ = "RealistikDash"
 __version__ = "v2.0.0"
 
-
-@dataclass
-class JsonFile:
-    file_name: str
-    file: dict = field(default_factory=dict)
-
-    def __post_init__(self) -> None:
-        """Reloads the file fully into memory."""
-
-        with open(self.file_name) as f:
-            self.file = loads(f.read())
-
-    def get_file(self) -> dict:
-        """Returns the loaded JSON file as a dict.
-        Returns:
-            Contents of the file.
-        """
-        return self.file
-
-    def write_file(self, new_content: Union[dict, list]) -> None:
-        """Writes `new_content` to the target file.
-        Args:
-            new_content (dict, list): The new content that should be placed
-                within the file.
-        """
-
-        with open(self.file_name, "w") as f:
-            dump(new_content, f, indent=4)
-
-        self.file = new_content
-
+T = TypeVar("T")
 
 class ConfigReader:
     """A parent class meant for the easy management, updating and the creation
     of a configuration `JSON` file."""
+    
+    _json: dict
+    updated_keys: Optional[list[str]] = None
```

Comment:
being optional is kinda weird, i would just make it non-optional and an empty list and just add to it if keys are updated